utils/timeout.ts

```diff
@@ -7,3 +7,9 @@ export function timeout<T>(time: number, returnValue: T): Promise<T> {
 export function promiseWithTimeout<T>(promise: Promise<T>, time: number) {
   return Promise.race([promise, timeout(time, -1)]);
 }
+
+let previousId = -1;
+export function debounce(func: () => any, time: number) {
+  window.clearTimeout(previousId);
+  previousId = window.setTimeout(func, time);
```

Comment:
Because the function is called in an input `onChange`, so everytime the user is modifying the input, it'll clear the last timeout, and prevent the function call. So basically it'll run like so:
- First call:
    - `window.clearTimeout(-1)` -> does nothing
    - `previousId = window.setTimeout(func, time)` -> `previousId = 1`
- Second call before the timeout ends:
    - `window.clearTimeout(1)` -> cancel the timeout, and thus the function call
    - `previousId = window.setTimeout(func, time)` -> `previousId = 2`
- Function is not called after that, so the function can be called after the duration of the timeout